### PR TITLE
PHPStan (Level 4) on tests/

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | PHPStan (Level 4) on tests/
| Type?             | improvement
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Relative to #16471
| How to test?      | Github Actions are :green_circle: 

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/27141)

### CODE REVIEW

**Review by matthieu-rolland — DISMISSED** [review]

**Review by matthieu-rolland — COMMENTED** [review]
ah, seems like a unit test related to CLDR is failing @Progi1984

File: tests/Integration/Behaviour/Features/Context/CustomerManagerFeatureContext.php [hunk 1]
```diff
@@ -272,8 +272,6 @@ public function assertCustomerWasNotFound($customerReference)
         } catch (CustomerNotFoundException $e) {
             return;
         }
-
-        throw new \Exception('Customer exists');
```
Comment (jolelievre) [inline tests/Integration/Behaviour/Features/Context/CustomerManagerFeatureContext.php:276]: So the test fails when the exception is not caught I get that this piece of code is useless But then so is the `return` in the catch block

Alternatively, you could use a `$caughtException = null;` before the try/catch block and fill it with caught exception If `$caughtException` is null the test fails (and you can throw a RuntimeException or a NoExceptionAlthoughExpectedException)

**Review by jolelievre — CHANGES_REQUESTED** [review]
Overall ok, just a suggestion
However unit tests are failing for now

File: tests/Integration/Behaviour/Features/Context/CustomerManagerFeatureContext.php (hunk 1 above)
Comment (Progi1984) [reply tests/Integration/Behaviour/Features/Context/CustomerManagerFeatureContext.php:276]: Done

**Review by matthieu-rolland — APPROVED** [review]

**Review by sowbiba — APPROVED** [review]

**Review by jolelievre — APPROVED** [review]

**Comment by sowbiba** [comment]
Thanks @Progi1984

**Pull request merged**